front/app/components/SignUpIn/events.ts

```diff
@@ -9,6 +9,14 @@ enum events {
 
 // ---------
 
+function emitOpenSignUpInModal(metaData: ISignUpInMetaData | undefined) {
```

Comment:
To have some kind of check on metaData (and reduce some repetition), I've created this function. `closeSignUpInModal` type (<ISignUpInMetaData>) was also wrong, given it didn't include undefined) causing incomplete type issues down the line (in e.g. SignUpInModal, see below).

Guess there's a better way.

Reply:
It's a way to send events across the application with an observable. For example when I need to know in App/index.tsx that I need to close a modal when an event happens somewhere deeper in the code.

Didn't look into changing this mechanism indeed.